### PR TITLE
[JS sample] Add CoreBot with Application-Insights 

### PR DESCRIPTION
**Pull request by gasper-az**
## Proposed Changes
  - Add sample `21 - CoreBot with Application-Insights`, based on JS [Sample#13](https://github.com/microsoft/BotBuilder-Samples/tree/master/samples/javascript_nodejs/13.core-bot)

## More Information
This sample:
- Uses [LUIS](https://www.luis.ai) to implement core AI capabilities
- Implements a multi-turn conversation using Dialogs
- Handles user interruptions for such things as `Help` or `Cancel`
- Prompts for and validates requests for information from the user
- Uses [Application Insights](https://docs.microsoft.com/azure/azure-monitor/app/cloudservices) to monitor your bot

## Testing
To test this sample:
1. Clone this branch
1. In a terminal, navigate to `samples/javascript_nodejs/13.core-bot`
1. Install modules with `npm install`
1. Setup LUIS
1. Run the sample with `npm start`

### CODE REVIEW

File: samples/javascript_nodejs/21.corebot-app-insights/bots/dialogAndWelcomeBot.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+     * @param {UserState} userState 
+     * @param {Dialog} dialog 
+     */
+    constructor(conversationState, userState, dialog) {
+        super(conversationState, userState, dialog);
+
+        this.onMembersAdded(async (turnContext, next) => {
+            const membersAdded = turnContext.activity.membersAdded;
+
+            for (let pos = 0; pos < membersAdded.length; pos++) {
```
Comment (Aliandi) [inline samples/javascript_nodejs/21.corebot-app-insights/bots/dialogAndWelcomeBot.js:20]: Couldn't this be done with a `foreach`?

**Review by Aliandi — APPROVED** [review]

File: samples/javascript_nodejs/21.corebot-app-insights/dialogs/mainDialog.js [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+            const didntUnderstandMessageText = `Sorry, I didn't get that. Please try asking in a different way (intent was ${ LuisRecognizer.topIntent(luisResult) })`;
+            await stepContext.context.sendActivity(didntUnderstandMessageText, didntUnderstandMessageText, InputHints.IgnoringInput);
+        }
+
+        return await stepContext.next();
+    }
+
+    /**
+     * Shows a warning if the requested From or To cities are recognized as entities but they are not in the Airport entity list.
+     * In some cases LUIS will recognize the From and To composite entities as a valid cities but the From and To Airport values
```
Comment (ceciliaavila) [inline samples/javascript_nodejs/21.corebot-app-insights/dialogs/mainDialog.js]: Please remove the 'a' in 

> LUIS will recognize the From and To composite entities as **a** valid cities

**Review by ceciliaavila — CHANGES_REQUESTED** [review]

File: samples/javascript_nodejs/21.corebot-app-insights/bots/dialogAndWelcomeBot.js (hunk 1 above)
Comment (gasper-az) [reply samples/javascript_nodejs/21.corebot-app-insights/bots/dialogAndWelcomeBot.js:20]: Yes, but AFAIK, the `for` loop is more performant than the `foreach` loop ([src](https://stackoverflow.com/a/43032526))

File: samples/javascript_nodejs/21.corebot-app-insights/dialogs/mainDialog.js (hunk 2 above)
Comment (gasper-az) [reply samples/javascript_nodejs/21.corebot-app-insights/dialogs/mainDialog.js]: Typo fixed

**Review by ceciliaavila — APPROVED** [review]

**Pull request closed without merging**